test/OrchardCore.Commerce.Tests/Fakes/FakeCartStorage.cs

```diff
@@ -30,4 +30,10 @@ public Task StoreAsync(ShoppingCart items)
         _carts[items.Id ?? string.Empty] = new ShoppingCart(items.Items);
         return Task.CompletedTask;
     }
+
+    public Task RemoveAsync(string shoppingCartId)
+    {
+        _carts.Remove(shoppingCartId ?? string.Empty);
```

Comment:
Is the `?? string.Empty` necessary? Internally we already use `string.IsNullOrEmpty` in the `GetCacheId`.
```suggestion
        _carts.Remove(shoppingCartId);
```